pkg/scale/varying_data_type.go

```diff
@@ -66,8 +66,11 @@ func (vdt *VaryingDataType) Set(value VaryingDataTypeValue) (err error) {
 }
 
 // Value returns value stored in vdt
-func (vdt *VaryingDataType) Value() VaryingDataTypeValue {
-	return vdt.value
+func (vdt *VaryingDataType) Value() (VaryingDataTypeValue, error) {
+	if vdt.value == nil {
+		return nil, fmt.Errorf("vdt value is nil")
+	}
```

Comment:
Noob questions on that:

1. Shouldn't we have it as a panic if the value is left unset? (Is it a programming only error, or can it be caused by user input/io etc.)
2. Is having a varying data type without a value set not allowed?

Reply:
Unset VDTs are used when decoding.  You supply an unset VDT as a destination.

Reply:
Thanks! So then shouldn't the caller handle the `nil` value and return an error if it shouldn't be nil? 🤔 
Or should it never be nil anywhere? 🤔

Reply:
IMO I dont think the caller should have the burden having to remember to do the nil check. As is what happened in my last PR, tim had to request changes to add the nil check. It feels more intuitive to me to return error from scale, that the caller then check because you always error check, and then handle as it sees fit

Reply:
That also makes sense, I guess if it's a sentinel error, one could always programmatically `errors.Is` the error and allow it to be nil if needed. I'm just wondering really, I'm not really for or against one way or another, so all good. Resolving this!